### ISSUE TITLE
Question: Can we do this without argo

### ISSUE THREAD

**Issue by zimab** [APGGroeiFabriek/PIVT#16]
If I change the templates(_EX: hlf-kube/peer-org-flow/templates/peerorg-workflow.yaml_)  to not use argo and kind type workflow will the process still work/what would I need to change? Our K8s environment does not allow argo currently. This would be for **samples/simple** example.

**Comment by raftAtGit** [comment]
nope, sorry, I'm afraid not!

you need a mechanism to orchestrate the tasks, (i.e. what will run after what), you may search for another tool, but:
1. it will be an almost complete re-write
2. most possibly you would need to install that tool to your K8S too. if you are not allowed to install Argo, then possibly you wont be allowed to install that tool either

**Comment by zimab** [comment]
@raftAtGit Thank you for the info.

**Comment by zimab** [comment]
is argo used in this network startup command?
`helm install ./hlf-kube --name hlf-kube -f samples/simple/network.yaml -f samples/simple/crypto-config.yaml` 

or is it just used in the next steps for channel and chaincode?

**Comment by raftAtGit** [comment]
no, it's not used when launching the network. but used in all flows.

it's not used behind the scenes, go over the samples in [README](https://github.com/APGGroeiFabriek/PIVT#go-over-samples), if there no call like `| argo submit` then it's not used.

**Comment by zimab** [comment]
so you mean that if I see `| argo submit` then it is used I assume, but not used in the above command I gave?

**Comment by raftAtGit** [comment]
yup, sorry, typo :) fixed above too..

**Comment by zimab** [comment]
Thank you :)

**Comment by zimab** [comment]
I am getting the following errors when bringing up the sample/simple network.

`2019-10-27 17:52:27.678 UTC [msp] getPemMaterialFromDir -> WARN 001 Failed reading file /etc/hyperledger/fabric/msp/admincerts/cert.pem: no pem content for file /etc/hyperledger/fabric/msp/admincerts/cert.pem
2019-10-27 17:52:27.678 UTC [main] InitCmd -> ERRO 002 Cannot run peer because error when setting up MSP of type bccsp from directory /etc/hyperledger/fabric/msp: Setup error: nil conf reference`

**Comment by raftAtGit** [comment]
do not use cryptogen 1.4.3. it's a known [issue](issues/5). also mentioned in requirements in README

**Comment by zimab** [comment]
**cryptogen:
 Version: 1.3.0**
 Commit SHA: ab0a67a
 Go version: go1.10.4
 OS/Arch: darwin/amd64

**Comment by raftAtGit** [comment]
himm, what is the output of this command?
`tree samples/simple/crypto-config/peerOrganizations/atlantis.com/peers/peer0.atlantis.com/msp/`

**Comment by zimab** [comment]
tree samples/simple/crypto-config/peerOrganizations/***/peers/peer0.***/msp/
samples/simple/crypto-config/peerOrganizations/***/peers/peer0.***/msp/
├── cacerts
│   └── ca.***-cert.pem
├── config.yaml
├── keystore
│   └── 320326145b205359bb43a6bf262a1c6a08c3bd407ae349c7653b5e8886300d6e_sk
├── signcerts
│   └── peer0.***-cert.pem
└── tlscacerts
    └── tlsca.***-cert.pem

4 directories, 5 files

**Comment by raftAtGit** [comment]
there should be an `admincerts` folder there too. never tried `cryptogen 1.3.0` before. looks like it doesnt create that folder. use 1.4.2 or 1.4.1, they are known to be working well

**Comment by zimab** [comment]
How can I upgrade cryptogen? I believe I have it from hyperledger binaries download

**Comment by raftAtGit** [comment]
follow the link in README/requirements

**Comment by zimab** [comment]
Thanks!

**Comment by zimab** [comment]
Can I remove hlf-kube/requirements.yaml since sample/simple does not use kafka currently? I want to be able to run the network without doing the following for sample/simple
`helm repo add kafka http://storage.googleapis.com/kubernetes-charts-incubator
helm dependency update ./hlf-kube/`

**Comment by raftAtGit** [comment]
I guess it should be fine

**Comment by zimab** [comment]
Since I cannot use argo I assume I have to run the commands manually for the channel/chaincode. Do I have the correct info below since some commands need to be run on only one and some need to be run on all peers? Anything I am missing?

create channel on peer1
all peers join channel
install chaincode on each peer
instantiate chaincode on only one peer
query on each peer
invoke on only one peer

**Comment by raftAtGit** [comment]
right, that's basically the flow. except:

query on each peer -> we do not do this
invoke on only one peer -> we do this for each peer. not mandatory but handy, see FAQ page for the reasoning

**Comment by zimab** [comment]
Thank you! and seems no need to do anchor peers commands(as basic fabric-samples byfn does)?

**Comment by raftAtGit** [comment]
I guess it's good practice actually to set anchor peers. but if you only have one peer per organization, possibly no need

**Comment by zimab** [comment]
I keep getting this error on only peer images..

PEER IMAGE ERRORS
Error from server (BadRequest): a container name must be specified for pod hlf-peer--ctdi--peer0-0, choose one of: [peer couchdb]

**Comment by raftAtGit** [comment]
well specify the container name as the error says ;)
`-c peer`

**Comment by zimab** [comment]
Where to specify that? in what command?

**Comment by raftAtGit** [comment]
while executing which command you get this error?

**Comment by zimab** [comment]
helm install ./hlf-kube --name hlf-kube -f samples/simple/network.yaml -f samples/simple/crypto-config.yaml

then the containers show error so I check logs and it shows the error provided

**Comment by raftAtGit** [comment]
so you try to check the logs with `kubectl` command, provide `-c peer` argument to `kubectl` command :)

man, sorry but this is not a Kubernetes basics forum!

**Comment by zimab** [comment]
oh wow thought that was container run error, sry

**Comment by idanilt** [comment]
I didn't liked argo, I used helm hooks (https://helm.sh/docs/topics/charts_hooks/) for install CC and join channel

**Comment by sillysachin** [comment]
> I didn't liked argo, I used helm hooks (https://helm.sh/docs/topics/charts_hooks/) for install CC and join channel

Can you share your approach in a blog or comment here?
